tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+
+import 'environment.dart';
+
+/// A function that returns true or false when given a [BuildConfig] and its
+/// name.
+typedef ConfigFilter = bool Function(String name, BuildConfig config);
+
+/// A function that returns true or false when given a [BuildConfig] name
+/// and a [GlobalBuild].
+typedef BuildFilter = bool Function(String configName, GlobalBuild build);
+
+/// Returns a filtered copy of [input] filtering out configs where test
+/// returns false.
+Map<String, BuildConfig> filterBuildConfigs(
+    Map<String, BuildConfig> input, ConfigFilter test) {
+  return {
+    for (final entry in input.entries)
+      if (test(entry.key, entry.value)) entry.key: entry.value,
+  };
+}
+
+/// Returns a copy of [input] filtering out configs that are not runnable
+/// on the current platform.
+Map<String, BuildConfig> runnableBuildConfigs(
+    Environment env, Map<String, BuildConfig> input) {
+  return filterBuildConfigs(input, (String name, BuildConfig config) {
+    return config.canRunOn(env.platform);
+  });
+}
+
+List<GlobalBuild> filterBuilds(
+    Map<String, BuildConfig> input, BuildFilter test) {
+  return [
+    for (final entry in input.entries)
+      for (final entry2 in entry.value.builds)
+        if (test(entry.key, entry2)) entry2,
+  ];
+}
+
+/// Returns a list of runnable builds.
+List<GlobalBuild> runnableBuilds(
+    Environment env, Map<String, BuildConfig> input) {
+  return filterBuilds(input, (String configName, GlobalBuild build) {
+    if (configName.contains('clang_tidy')) {
+      // TODO(johnmccutchan): Figure out how we should manage duplicate build
```

Comment:
The builds that `canRunOn` a platform should all be uniquely named, now.

Reply:
Removed.